scripts/imgtool/keys/ecdsa.py

```diff
@@ -61,6 +61,8 @@ def sig_len(self):
         return 72
 
     def verify(self, signature, payload):
+        # strip possible paddings added during sign
+        signature = signature[:signature[1] + 2]
```

Comment:
I don't quite see how this will work, the length in ASN.1 is a number of elements, not a number of bytes.

Reply:
I'm not an expert of ASN.1. Per my knowledge, the second byte (or more, if bit-7 of first byte is set) is length in the case of OCTET STRING.

I have dumped a signature and the observation is the same. With regard to the padding option (default adding it, or not adding on user's request) in future versions, verify function in imgtool.py still need to be functional for both of them. No matter what is supported in MCUboot boot loader in actual embedded systems.

Reply:
For example, an ECDSA signature I've added is:
304502206961ebb1....
30 45 (30=SEQUENCE, 45=69 bytes)
   02 20 (INTEGER 32 bytes)
      69 61 eb b1 ....

Reply:
The signature is an ASN.1 sequence, not just an octet string. The length is encoded as you mentioned (in octets, I was getting confused working with CBOR recently).